src/sentry/spans/buffer.py

```diff
@@ -310,6 +320,37 @@ def _group_by_parent(self, spans: Sequence[Span]) -> dict[tuple[str, str], list[
 
         return trees
 
+    def _compress_span_payloads(self, payloads: list[bytes]) -> bytes:
+        combined = b"\x00".join(payloads)
+
+        # If compression is disabled, return the combined data as-is
+        if self._zstd_compressor is None:
+            return combined
```

Comment:
This might be a problem if we want to roll back. Could we instead write the payloads separately in this  case?